### PR TITLE
Separate audit checks tests

### PR DESCRIPTION
**Pull request by kaligrafy**
## Summary by CodeRabbit

* **Documentation**
  * Updated audit checks guide with unified naming, expanded examples, test organization guidance, and added JSDoc for a language check.

* **Tests**
  * Added many new unit and integration audit tests, reusable test helpers, service-level mocks, migrated suites to integration-style runners, and improved edge-case coverage; removed stray logging.

* **Localization**
  * Cleaned up whitespace/indentation in audit translation files (no text changes).

* **Chores**
  * Tightened audit check return types, switched to type-only imports, simplified audit aggregation logic, and adjusted public test-facing exports.

### CODE REVIEW

File: packages/evolution-backend/src/services/audits/auditChecks/checks/__tests__/VisitedPlaceAuditChecks.test.ts [hunk 1]
```diff
@@ -7,267 +7,93 @@
...
                 ignore: false
-            });
-        });
-
-        it('should pass when geography is missing (handled by VP_M_Geography)', () => {
-            const context = createFullContext({
-                visitedPlace: createMockVisitedPlace({ geography: undefined })
-            });
+            })
+        };
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-backend/src/services/audits/auditChecks/checks/__tests__/VisitedPlaceAuditChecks.test.ts]: fixed all your suggestion, please verify I did not forget anything.

File: packages/evolution-backend/src/services/audits/auditChecks/checks/__tests__/journey/J_M_StartDate.test.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        expect(result).toEqual({
+            objectType: 'journey',
+            objectUuid: validUuid,
+            errorCode: 'J_M_StartDate',
+            version: 1,
+            level: 'error',
+            message: 'Journey start date is missing',
+            ignore: false
+        });
+    });
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-backend/src/services/audits/auditChecks/checks/__tests__/journey/J_M_StartDate.test.ts]: added

File: packages/evolution-backend/src/services/audits/auditChecks/checks/README.md [hunk 3]
```diff
@@ -4,52 +4,36 @@ This directory contains the actual audit check functions organized by survey obj
...
 
-All audit check functions follow the pattern: `[SURVEY_OBJECT]_[TYPE]_[Description]`
-
-- **Entity prefixes**: `I` (Interview), `HH` (Household), `HM` (Home), `P` (Person), `J` (Journey), `VP` (VisitedPlace), `T` (Trip), `S` (Segment)
-- **Audit types**: `I` (Invalid), `L` (Logical), `M` (Missing)
-
-## Current Audit Checks
-
-### Missing Data Checks (M)
+All audit check functions follow the pattern: `[OBJECT]_[TYPE]_[Description]`
```
Comment (tahini) [inline packages/evolution-backend/src/services/audits/auditChecks/checks/README.md]: Pour que ça matche clairement avec les section plus bas `[OBJECTPREFIX]_[CHECKTYPE]_[Description]`

File: packages/evolution-backend/src/services/audits/auditChecks/checks/__tests__/household/testHelper.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+export const createMockHousehold = (overrides: Partial<Household> = {}, validUuid = uuidV4()) => {
+    return {
+        _uuid: validUuid,
+        size: 3,
+        ...overrides
+    } as Household;
+};
+
+export const createContext = (householdOverrides: Partial<Household> = {}, validUuid = uuidV4()): HouseholdAuditCheckContext => {
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-backend/src/services/audits/auditChecks/checks/__tests__/household/testHelper.ts]: fixed

File: packages/evolution-backend/src/services/audits/auditChecks/checks/__tests__/JourneyAuditChecks.test.ts [hunk 5]
```diff
@@ -6,157 +6,70 @@
...
+            TEST_FAIL_2: (): AuditForObject => ({
+                objectType: 'journey',
+                objectUuid: validUuid,
+                errorCode: 'TEST_FAIL_2',
+                version: 1,
+                level: 'warning',
+                message: 'Test failure 2',
+                ignore: false
+            })
+        };
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-backend/src/services/audits/auditChecks/checks/__tests__/JourneyAuditChecks.test.ts]: fixed

[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-backend/src/services/audits/auditChecks/checks/__tests__/JourneyAuditChecks.test.ts]: Removed all Partial for AuditObject, clearer and less error prone
